Tools/clinic/clinic.py

```diff
@@ -3204,6 +3201,8 @@ def converter_init(self, *, accept={str}, encoding=None, zeroes=False):
             # sorry, clinic can't support preallocated buffers
             # for es# and et#
             self.c_default = "NULL"
+        if NoneType in accept and self.c_default == "Py_None":
```

Comment:
is this really okay?  we hold the GIL and the code you've modified at a quick first glance isn't _using_ the value when it is == Py_None, but we never Py_INCREF'd the Py_None and there is nothing preventing us from treating it as if we had.

It also potentially adds situations where None is accepted as a parameter where it wasn't before?  I'd need to go audit all of the APIs this change is trying to update to see if each of them intends to allow None as a value for a review of that.  In that case if someone _did_ pass in None from Python code, it should've been Py_INCREF'd - who's doing the DECREF before clobbering it?

Reply:
This is the `str` converter. The target's type is `const char *`. Py_None is invalid value here. None is converted to NULL. For example see `_io.open` and various encode/decode methods.